### PR TITLE
topology2: cavs-sdw: Enable PDM 1 mics for 4ch

### PR DESCRIPTION
**Pull request by ranj063**
It would be better to make these dependedent on NUM_DMICS. Something to improve in a follow up PR.

### CODE REVIEW

File: tools/topology/topology2/sof-ace-tplg/CMakeLists.txt
Comment (plbossart) [inline tools/topology/topology2/sof-ace-tplg/CMakeLists.txt:21]: The PCM IDs seem weird, and we've never exposed the second DMIC device

````
ubuntu@sh-adlp-rvp-sdw-03:~$ arecord -l
**** List of CAPTURE Hardware Devices ****
card 0: sofsoundwire [sof-soundwire], device 1: Jack in (*) []
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 0: sofsoundwire [sof-soundwire], device 10: DMIC (*) []
  Subdevices: 1/1
  Subdevice #0: subdevice #0
````

the list of PCM devices we use for SoundWire devices is this:

````
PCM0   playback	Jack Out
PCM1   capture 	Jack In
PCM2   playback	Speaker
PCM3   capture	Amplifier Reference (typically based on I/V feedback)
PCM4   capture 	Microphone
PCM5   playback	HDMI 1
PCM6   playback	HDMI 2
PCM7   playback	HDMI 3
PCM8   playback	HDMI 4
PCM9   RESERVED
PCM10  capture	DMIC
PCM11  capture	DMIC16kHz
PCM12  capture 	BufferedMic (aka KPB)
PCM13  playback Bluetooth
PCM14  capture 	Bluetooth

````

**Review by plbossart — CHANGES_REQUESTED** [review]
NAK on PCM devices

File: tools/topology/topology2/sof-ace-tplg/CMakeLists.txt
Comment (ranj063) [reply tools/topology/topology2/sof-ace-tplg/CMakeLists.txt:21]: @plbossart I cannot change the PCM ID or the second DMIC in this PR. This is simply to make sure that the PCM1 mics are enabled for 4ch capture

Comment (plbossart) [reply tools/topology/topology2/sof-ace-tplg/CMakeLists.txt:21]: Why can't you change those PCM IDs @ranj063 ? They are not quite right and incompatible with existing platforms.
You can do this in a second patch if you want to, but the existing result is just wrong as it stands.

Comment (ranj063) [reply tools/topology/topology2/sof-ace-tplg/CMakeLists.txt:21]: @plbossart im confused. what is the exact change you're looking for? The PCM ID's look correct to me based on your lsit no? PCM 1 for Jack 1 and PCM10 for DMIC

Comment (plbossart) [reply tools/topology/topology2/sof-ace-tplg/CMakeLists.txt:21]: this is what I object to:
line 21 'DMIC0_ID=2,DMIC1_ID=3'

Comment (ranj063) [reply tools/topology/topology2/sof-ace-tplg/CMakeLists.txt:21]: But those arent the PCM ID's. @bardliao these are the link IDs in the machine driver isnt it?

Comment (bardliao) [reply tools/topology/topology2/sof-ace-tplg/CMakeLists.txt:21]: > But those arent the PCM ID's. @bardliao these are the link IDs in the machine driver isnt it?

Yes, they are dai link IDs. The pcm IDs are DMIC0_PCM_ID, etc.

Comment (plbossart) [reply tools/topology/topology2/sof-ace-tplg/CMakeLists.txt:21]: Wow, now I am completely confused, how can the values 2 and 3 work dailink indices of the DMICs for SoundWire platforms.

What a mess.

Comment (bardliao) [reply tools/topology/topology2/sof-ace-tplg/CMakeLists.txt:21]: @plbossart sof-mtl-rt711-4ch is for MTL RVP which doesn't have amp. I.e. Only rt711 is connected. So, the DMIC dai link IDs starts from 2.

Comment (plbossart) [reply tools/topology/topology2/sof-ace-tplg/CMakeLists.txt:21]: with topology1 the SoundWire + DMIC topologies were in a different file. Looks like for topology2 we are using a single file, so no wonder things are confusing - different conventions are used.

It's a mess.

Comment (lgirdwood) [reply tools/topology/topology2/sof-ace-tplg/CMakeLists.txt:21]: Why are not using teh same conventions and link IDs as before ? Whats the userspace impact if we align them ?

**Comment by lgirdwood** [comment]
@ranj063 any alignment yet ?

**Comment by kv2019i** [comment]
@ranj063 Status? No update for a month, please mark as draft if not ready to go yet.

**Comment by ranj063** [comment]
> @ranj063 any alignment yet ?

@lgirdwood there was a conflict. Fixed npw. @plbossart is good with the change. Waiting for his approval

**Comment by ranj063** [comment]
> @ranj063 Status? No update for a month, please mark as draft if not ready to go yet.

This change is correct. The only reason it is not critical is because we use the BIOS NHLT in the SDW case.

**Review by plbossart — APPROVED** [review]

**Comment by kv2019i** [comment]
Our familiar DSP panic shows up on one DUT in https://sof-ci.01.org/sofpr/PR7252/build6829/devicetest/index.html , but this is known bug in baseline. Proceeding with merge.

**Pull request merged**
